relay/sources/relayd/src/remote_run.rs

```diff
@@ -65,32 +65,66 @@ impl RemoteRun {
         })
     }
 
+    pub fn target_to_vector(&self, job_config: Arc<JobConfig>) -> Vec<String> {
+        match &self.target {
+            RemoteRunTarget::All => job_config
+                .nodes
+                .read()
+                .expect("Cannot read nodes list")
+                .get_neighbors_from_target(RemoteRunTarget::All),
+            RemoteRunTarget::Nodes(nodeslist) => nodeslist.clone(),
+        }
+    }
+
     pub fn run(
         &self,
         job_config: Arc<JobConfig>,
     ) -> Result<impl warp::reply::Reply, warp::reject::Rejection> {
+        if self.run_parameters.keep_output && self.run_parameters.asynchronous {
+            return Err(warp::reject::custom(
+                "keep_output and asynchronous cannot be true simultaneously",
+            ));
+        }
         if self.target == RemoteRunTarget::All {
             info!("conditions OK");
```

Comment:
this is probably useless, if there is no error we can assume parameters were correct

Reply:
up